src/radical/saga/adaptors/pbspro/pbsprojob.py

```diff
@@ -298,59 +246,59 @@ def _script_generator(url, logger, jd, ppn, gres, version, is_cray=False,
-    # https://github.com/radical-cybertools/radical.saga/issues/406
-    #
-    if jd.candidate_hosts:
-        if 'BIG_FLASH' in jd.candidate_hosts:
-            node_properties.append('bigflash')
-        elif len(jd.candidate_hosts) == 1:
-            queue_spec += '@%s' % jd.candidate_hosts[0]
-        else:
-            raise rse.NotImplemented("unsupported candidate_hosts [%s]"
-                                      % jd.candidate_hosts)
-
-    if 'cheyenne'  in url.host.lower() or \
-       'cheyenne'  in os.uname()[1].lower():
-        is_cheyenne = True
+    # TODO: Special cases for PBS/TORQUE on Cray
+    if any(v in version for v in ['PBSPro_10', '4.2.7']):       # edison, hopper
+        pbs_params += '#PBS -l mppwidth=%s\n' % jd.total_cpu_count
+    elif any(v in version for v in ['PBSPro_12', 'PBSPro_13']): # archer
+        pbs_params += '#PBS -l select=%d\n' % nnodes
+    elif is_cray:                                               # kraken, jaguar
+        pbs_params += '#PBS -l size=%s\n' % jd.total_cpu_count
+    elif node_properties:
+        pbs_params += '#PBS -l nodes=%d:ppn=%d%s\n' % (
+            nnodes, ppn, ''.join([':%s' % prop for prop in node_properties]))
     else:
-        is_cheyenne = False
-
-    # TODO: Special cases for PBS/TORQUE on Cray. Different PBSes,
-    #       different flags. A complete nightmare...
-    #       The more we add, the more it screams for a refactoring
-    if   is_cheyenne                     : pbs_params += "#PBS -l select=%d:ncpus=%d\n" % (nnodes, 36)           # cheyenne, ppn=36
-    elif 'PBSPro_10'       in version: pbs_params += "#PBS -l mppwidth=%s \n"       % jd.total_cpu_count     # hopper
-    elif 'PBSPro_12'       in version: pbs_params += "#PBS -l select=%d\n"          % nnodes                 # pbspro 12, archer
-    elif 'PBSPro_13'       in version: pbs_params += "#PBS -l select=%d\n"          % nnodes                 # pbspro 13
-    elif '4.2.6'           in version: pbs_params += "#PBS -l nodes=%d\n"           % nnodes                 # titan
-    elif '4.2.7'           in version: pbs_params += "#PBS -l mppwidth=%s \n"       % jd.total_cpu_count     # edison, hopper
-    elif 'Version: 5.'     in version: pbs_params += "#PBS -l procs=%d\n"           % jd.total_cpu_count     # torqye 5
-    elif 'version: 2.3.13' in version: pbs_params += "#PBS -l ncpus=%d\n"           % ncpus                  # blacklight
-    elif '14.2'            in version: pbs_params += "#PBS -l select=%d:ncpus=%d\n" % (nnodes, ppn)          # pbspro 14
-    elif is_cray                     : pbs_params += "#PBS -l size=%s\n"            % jd.total_cpu_count     # kraken, jaguar
-    else                             : pbs_params += "#PBS -l nodes=%d:ppn=%d%s\n"  % (nnodes, ppn,          # default
-                                       ''.join([':%s' % prop for prop in node_properties]))
+        pbs_params += '#PBS -l select=%d:ncpus=%d\n' % (nnodes, ppn)
```

Comment:
This refactoring seems to have lost several version switches, if I am not mistaken (`'Version: 5.'` being a random example).  What is the rationale?  Cleanup of versions you would consider not be used anymore?  How would we know?

Reply:
I cleaned out parts related to systems that were obsolete for awhile, and yes, considering that that versions are outdated. Regarding `'Version: 5.'` as I understood it is related mostly to Torque, which also uses `PBS` directives. So the original goal was to remove outdated-outdated ones

Reply:
That's fair.  If push comes to shove, we have the versions in git history :-D